### DISCUSSION TITLE
.gitmodules for a repo with embedded datasets

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#287] [Q&A] [answered]
Are .gitmodules generated when we do `datalad run-procedure` or should they be generated by hand? If the latter,
	datalad-id = 6f880209-b5aa-4249-ae5f-e1b6fe4d4796

is the version of the dataset, right?

Also, is it a good idea to always have them?

**Comment by lucasgautheron** [comment, marked as answer]
.gitmodules are neither generated by `datalad run-procedure` or by hand -- they are only affected by `git submodule` or `datalad install` operations.

I think it is safe to say that they should NEVER be updated by hand (only through the appropriate git submodule/datalad commands)

Re IDs, I think the best explanation is here: https://docs.datalad.org/en/stable/basics.html#types-of-ids-used-in-datasets

My comment is probably not really helpful, so if I may ask -- why this question? what are you trying to do?

**Reply by alecristia** [reply]
it IS helpful! I just edited a set of instructions where I was saying that people should copy over .gitmodules (which I did in the past, and it caused headaches).

I also say there to copy over .gitattributes -- is that an error too?

**Comment by lucasgautheron** [comment]
.gitattributes is safe to copy :)


Le mer. 15 sept. 2021 à 18:46, Alex Cristia ***@***.***> a
écrit :

> it IS helpful! I just edited a set of instructions where I was saying that
> people should copy over .gitmodules (which I did in the past, and it caused
> headaches).
>
> I also say there to copy over .gitattributes -- is that an error too?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/LAAC-LSCP/ChildProject/discussions/287#discussioncomment-1334554>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABI4QHSKMI7S3NAYIWS7BIDUCDEU5ANCNFSM5EC2DT7A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Lucas Gautheron
06 79 23 86 47

Laboratoire de Sciences Cognitives et Psycholinguistique
Bureau 414 - 29 rue d'Ulm
75005 Paris
